### ISSUE TITLE
pylint compatibility + pycharm intellisense compatibility

### ISSUE THREAD

**Issue by cre4ture** [okken/pytest-check#175]
Hello,

I found `pytest-check` and think its quite useful for us.
We are using pylint for the static code analysis of our python code and tests.
`pytest-check` introduces false positives for pylint like these:

```
from pytest_check import check

...

check.equal(response.status.code, ERROR_CODE_SUCCESSFUL)
```

```
E1101: Instance of 'CheckContextManager' has no 'equal' member (no-member)
```

I know I can suppress this false positive, but I'd prefer to not have to do this.
Also, the intellisense of PyCharm seems to have an issue as well as it doesn't report `equal` or related member functions.

I know that this is not necessarily an issue of `pytest-check` but rather PyLint or PyCharm.
But anyway, since this tool is supposed to be used widely (right?) I thought there should be a standard recommended solution available that is different to just suppression of the false positives.

What do you think?

**Comment by cre4ture** [comment]
For now I gave up and returned to the use of simple asserts.
I don't see myself as an python expert. 
I have too less experience with it to come up with an own solution.

But if you have a concrete idea how to solve this I could spend a bit time to work on it.

**Comment by okken** [comment]
Just tested this with a small example with both:
- pylint
- ruff check (with all rules turned on)

Ruff doesn't find any problems.
pylint does. I think it's really a problem with pylint.
However, pytest-check does some non-standard modifications of the `check` object that wouldn't surprise me to trip up some linters. 
My suggestion would be to switch linters to `ruff` or `black --check`

Here's my example.

**pyproject.toml**
```
[tool.ruff.lint]
select = ["ALL"]
ignore = ["D203", "D212"]
```
Note: ignores are there because D203 is incompatible with D211, and D212 is incompatible with D213.

**test_equal.py**
```
"""Test the `check.equal` function."""

from pytest_check import check


def test_equal() -> None:
    """Test that 1 is equal to 1."""
    check.equal(1, 1)
```

**ruff**
```
$ ruff check test_equal.py
All checks passed!
```

**pylint**
```
$ pylint test_equal.py      
************* Module test_equal
test_equal.py:8:4: E1101: Instance of 'CheckContextManager' has no 'equal' member (no-member)
```

## Conclusion
I think it's an issue with pylint, as ruff seems to figure out what's going on ok. 
Even if it isn't a problem with pylint, I can't justify attempting to understand pylint enough to fix it.  

Closing this, but open to contributions if anyone else wants to take a crack at it.
